run_checkm_bakta.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+#SBATCH -J 500_ncbi_redgenes
+#SBATCH -p short
+#SBATCH -t 20:00:00
+#SBATCH -N 1
+#SBATCH -c 4
+#SBATCH --mem 64g
+#SBATCH -o /panfs/roles/redgenes/slurm-%x-%A-%a-%j-%N.out
+#SBATCH -e /panfs/roles/redgenes/slurm-%x-%A-%a-%j-%N.err
+#SBATCH --export ALL
+#SBATCH --mail-type=ALL,TIME_LIMIT_50,TIME_LIMIT_90
+#SBATCH --mail-user=roles@health.ucsd.edu
+
+set -x
+set -e
+set -o pipefail
+
+source /home/roles/anaconda/bin/activate
+conda activate checkm_bakta
+
+# Check input parameters
+if [ "$#" -ne 3 ]; then
+    echo "Usage: $0 <input_dir_with_fasta_files> <output_dir> <number_of_parallel_jobs>"
+    exit 1
+fi
+
+input_dir=$1
+output_dir=$2
+parallel_jobs=$3
+
+# Create output directory if it doesn't exist
+mkdir -p "$output_dir"
+
+# Function to process each FASTA file
+process_fasta() {
+    fasta_file=$1
+    base_name=$(basename "$fasta_file" .fasta)
+
+    # Create directory for each fasta file in the specified output directory
+    sample_dir="$output_dir/$base_name"
+    mkdir -p "$sample_dir/checkm"
+    mkdir -p "$sample_dir/bakta"
+
+    # Run CheckM
+    checkm lineage_wf --tab_table -x fasta "$fasta_file" "$sample_dir/checkm" > "$sample_dir/checkm/lineage.log"
+    if [ $? -ne 0 ]; then
+        echo "CheckM failed for $fasta_file"
+        return 1
```

Comment:
I had a weird checkm index error which did not get caught by ` if [ $? -ne 0 ]`:
```
*******Entering process_fasta
/projects/greengenes2/20240331_troubleshoot_kofamscan/data_ecoli/GCF_000005845.2_ASM584v2_genomic.fna
GCF_000005845.2_ASM584v2_genomic.fna
CheckM started...
Traceback (most recent call last):
  File "/home/y1weng/mambaforge/envs/gg2_bakta_db/bin/checkm", line 856, in <module>
    checkmParser.parseOptions(args)
  File "/home/y1weng/mambaforge/envs/gg2_bakta_db/lib/python3.9/site-packages/checkm/main.py", line 979, in parseOptions
    self.tree(options)
  File "/home/y1weng/mambaforge/envs/gg2_bakta_db/lib/python3.9/site-packages/checkm/main.py", line 141, in tree
    binFiles = self.binFiles(
  File "/home/y1weng/mambaforge/envs/gg2_bakta_db/lib/python3.9/site-packages/checkm/main.py", line 106, in binFiles
    binFile = files[1]
IndexError: list index out of range
```
This caused the bash script to stop instead of printing 'FAIL' to checkm_results.txt, which was the intended behavior I believe.

Trying to think on the spot maybe we can do something like
```
+e
checkm lineage_wf --tab_table -x fasta "$fasta_file" "$sample_dir/checkm" > "$sample_dir/checkm/lineage.log"
-e
```
then check checkm_results.txt and stops if the file is empty (or has some sort of indication where checkm failed)?